metadata-io/src/main/java/com/linkedin/metadata/search/utils/ESAccessControlUtil.java

```diff
@@ -0,0 +1,266 @@
+package com.linkedin.metadata.search.utils;
+
+import static com.linkedin.metadata.utils.SearchUtil.ES_INDEX_FIELD;
+import static com.linkedin.metadata.utils.SearchUtil.KEYWORD_SUFFIX;
+
+import com.datahub.authorization.AuthUtil;
+import com.datahub.authorization.ConjunctivePrivilegeGroup;
+import com.datahub.authorization.DisjunctivePrivilegeGroup;
+import com.datahub.authorization.EntitySpec;
+import com.google.common.collect.ImmutableList;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.template.StringArray;
+import com.linkedin.metadata.aspect.hooks.OwnerTypeMap;
+import com.linkedin.metadata.authorization.PoliciesConfig;
+import com.linkedin.metadata.models.registry.EntityRegistry;
+import com.linkedin.metadata.search.SearchEntity;
+import com.linkedin.metadata.search.SearchResult;
+import com.linkedin.metadata.timeseries.elastic.indexbuilder.MappingsBuilder;
+import com.linkedin.policy.DataHubActorFilter;
+import com.linkedin.policy.DataHubPolicyInfo;
+import com.linkedin.policy.PolicyMatchCriterion;
+import com.linkedin.policy.PolicyMatchCriterionArray;
+import io.datahubproject.metadata.context.ActorContext;
+import io.datahubproject.metadata.context.OperationContext;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.TermsQueryBuilder;
+
+@Slf4j
+public class ESAccessControlUtil {
+  private ESAccessControlUtil() {}
+
+  private static final String OWNER_TYPES_FIELD = "ownerTypes";
+  private static final QueryBuilder MATCH_ALL = QueryBuilders.matchAllQuery();
+  /*
+    These two privileges should be logically the same for search even
+    if one might allow search but not the entity page view at some point.
+  */
+  private static final Set<String> FILTER_PRIVILEGES =
+      Set.of(
+          PoliciesConfig.VIEW_ENTITY_PRIVILEGE.getType(),
+          PoliciesConfig.VIEW_ENTITY_PAGE_PRIVILEGE.getType());
```

Comment:
since we're treating these two privileges as identical right now, do you think it makes sense to update the frontend to allow users to see entity pages if they have the `VIEW_ENTITY_PRIVILEGE`? I could see it going either way - wanting to keep it as is and allow them to see things in search but not view the whole page through the UI would then be doable if we keep it.

i'm leaning towards keeping it as we have it - as far as our API goes, these two privileges are identical for now, but leave the functionality we have where if they don't have `VIEW_ENTITY_PAGE_PRIVILEGE` they get the unathorized page in the UI for their profile

Reply:
Yes, agree that for now they are considered identical. The actual values here was moved to the `PoliciesConfig` which includes similar collections of privileges. Also added a comment to that effect (that they'll be considered identical)